### PR TITLE
fix h2 regressions, h3 improvements

### PR DESCRIPTION
**Pull request by lpuskas**
* ListCharts response properly encoded not to loose information
* remove dependency between h2/h3 env resolver services
* repo update operation moved to the new interface (supported also on h3)
*  fixed list endpoints call

* extract factory method for the helm service facade and the unified releaser and create a high level deployment integration test to cover the releaser basic functionality
* extract the helm config to it's own package
* resolve default repo urls from code if none provided
* fix ark chart version

### CODE REVIEW

File: cmd/pipeline/main.go [hunk 1]
```diff
@@ -64,6 +64,10 @@ import (
 	zaplog "logur.dev/integration/zap"
 	"logur.dev/logur"
 
+	"github.com/banzaicloud/pipeline/internal/helm"
+
+	"github.com/banzaicloud/pipeline/internal/cmd"
```
Comment (sagikazarmark) [inline cmd/pipeline/main.go:69]: Import sorting

File: cmd/worker/main.go [hunk 2]
```diff
@@ -38,6 +37,10 @@ import (
 	zaplog "logur.dev/integration/zap"
 	"logur.dev/logur"
 
+	"github.com/banzaicloud/pipeline/internal/helm"
+
+	"github.com/banzaicloud/pipeline/internal/cmd"
```
Comment (sagikazarmark) [inline cmd/worker/main.go:42]: Import sorting

File: internal/cmd/config.go [hunk 3]
```diff
@@ -89,19 +90,17 @@ type Config struct {
...
-		Tiller struct {
-			Version string
-		}
-
-		Home string
-
-		Repositories map[string]string
+	Github struct {
+		Token string
+	}
```
Comment (sagikazarmark) [inline internal/cmd/config.go:95]: This was removed recently. Is this a bad merge conflict resolution?

File: internal/cmd/config.go [hunk 4]
```diff
@@ -89,19 +90,17 @@ type Config struct {
...
 
-		// flag signaling the helm version
-		Version string
+	Gitlab struct {
+		URL   string
+		Token string
```
Comment (sagikazarmark) [inline internal/cmd/config.go:99]: This was removed recently. Is this a bad merge conflict resolution?

File: internal/helm/envresolver.go [hunk 5]
```diff
@@ -64,60 +66,74 @@ type EnvResolver interface {
...
+		home:     path.Join(er.helmHomesDir, orgName, helmPostFix),
 		platform: false,
 	}, nil
 }
 
-func (h2r helm2EnvResolver) ResolvePlatformEnv(ctx context.Context) (HelmEnv, error) {
+func (er envResolver) ResolvePlatformEnv(ctx context.Context) (HelmEnv, error) {
 	return HelmEnv{
-		home:     path.Join(h2r.helmHomes, PlatformHelmHome, helmPostFix),
+		home:     path.Join(er.helmHomesDir, PlatformHelmHome, helmPostFix),
```
Comment (sagikazarmark) [inline internal/helm/envresolver.go:96]: What happens if this guy registers to the dev preview?

https://github.com/pipeline

File: internal/cmd/config.go (hunk 3 above)
Comment (pepov) [reply internal/cmd/config.go:95]: yes

File: internal/helm/envresolver.go (hunk 5 above)
Comment (pepov) [reply internal/helm/envresolver.go:96]: I will introduce a new configuration option for the platform home

Comment (pepov) [reply internal/helm/envresolver.go]: I chose a different approach: https://github.com/banzaicloud/pipeline/pull/2861/commits/6d66faf27d2067f035a9cb8ae4013e0e992bce2f

**Pull request merged**